### PR TITLE
Fix the missing cached image, add comments

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This change reduces CI running time from 9 to 7 minutes.

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -28,11 +28,18 @@ options:
     - DOCKER_CLIENT_TIMEOUT=120
     - COMPOSE_HTTP_TIMEOUT=120
 
+# Cache the testbase image in Container Regisrty, to be reused by subsequent
+# builds. The technique is described here:
+# https://cloud.google.com/cloud-build/docs/speeding-up-builds#using_a_cached_docker_image
+images: ['gcr.io/$PROJECT_ID/trillian_testbase:latest']
```
Comment (AlCutter) [inline cloudbuild.yaml:34]: Is there any potential for unfortunate interactions if there are multiple PRs being worked on at the same time?
I wasn't sure if this sort of thing was a problem or not, so my original plan was to just push the images on the merge-to-master trigger.

Comment (pavelkalinnikov) [reply cloudbuild.yaml:34]: I see. I'm not sure how "thread-safe" this is. I think, as long as one push wins another, it should be fine. This is just for caching.

Comment (pavelkalinnikov) [reply cloudbuild.yaml:34]: Pushing only when updating master seems fine as well.

Comment (pavelkalinnikov) [reply cloudbuild.yaml:34]: How about I add a TODO here?

Comment (AlCutter) [reply cloudbuild.yaml:34]: oke doke, sgtm!

**Review by AlCutter — APPROVED** [review]

**Pull request merged**